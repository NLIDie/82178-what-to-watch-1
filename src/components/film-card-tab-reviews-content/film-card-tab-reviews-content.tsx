import * as React from "react";

// Components
import {ReviewList} from "_Components/review-list/review-list";

// Utils
import {chooseEvenItems} from "_Utils/choose-even-items";
import {chooseOddItems} from "_Utils/choose-odd-items";

// Types
import {TProps} from "./film-card-tab-reviews-content.types";

export function FilmCardTabReviewsContent(props: TProps): JSX.Element {
  const {comments} = props;

  const firstColumnReviews = React.useMemo(() => {
    return chooseEvenItems(comments);
  }, [comments]);

  const secondColumnReviews = React.useMemo(() => {
    return chooseOddItems(comments);
  }, [comments]);

  return (
    <div className="movie-card__reviews movie-card__row">
      <div className="movie-card__reviews-col">
        <ReviewList reviews={firstColumnReviews} />
      </div>

      <div className="movie-card__reviews-col">
        <ReviewList reviews={secondColumnReviews} />
      </div>
    </div>
  );
}
