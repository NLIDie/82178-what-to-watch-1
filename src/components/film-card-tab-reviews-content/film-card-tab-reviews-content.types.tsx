// Types
import {TComment} from "_Types/comment";

export type TProps = {
  readonly comments: TComment[];
};
