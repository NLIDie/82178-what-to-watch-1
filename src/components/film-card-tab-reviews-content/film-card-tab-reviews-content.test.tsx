import * as React from "react";
import {create} from "react-test-renderer";

// Components
import {FilmCardTabReviewsContent} from "./film-card-tab-reviews-content";

describe(`Components/FilmCardTabReviewsContent`, () => {
  it(`Correct renderer the component`, () => {
    const tree = create(
        <FilmCardTabReviewsContent
          comments={[
            {
              comment: `comment 1`,
              date: new Date().toDateString(),
              id: 1,
              user: {
                id: 1,
                name: `kirill`
              },
              rating: 8.9
            }
          ]}
        />
    );

    expect(tree).toMatchSnapshot();
  });
});
