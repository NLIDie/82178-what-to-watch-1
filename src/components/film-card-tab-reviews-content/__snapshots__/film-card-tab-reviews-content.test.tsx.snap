// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Components/FilmCardTabReviewsContent Correct renderer the component 1`] = `
<div
  className="movie-card__reviews movie-card__row"
>
  <div
    className="movie-card__reviews-col"
  >
    <div
      className="review"
    >
      <blockquote
        className="review__quote"
      >
        <p
          className="review__text"
        >
          comment 1
        </p>
        <footer
          className="review__details"
        >
          <cite
            className="review__author"
          >
            kirill
          </cite>
          <time
            className="review__date"
            dateTime="Mon Jun 17 2019"
          >
            June
             
            17
            , 
            2019
          </time>
        </footer>
      </blockquote>
      <div
        className="review__rating"
      >
        8.9
      </div>
    </div>
  </div>
  <div
    className="movie-card__reviews-col"
  />
</div>
`;
